crypto/ocsp/ocsp_verify.c

```diff
@@ -377,6 +381,8 @@ int OCSP_basic_verify(OCSP_BASICRESP *bs, STACK_OF(X509) *certs,
 
 end:
   sk_X509_pop_free(chain, X509_free);
-  sk_X509_free(untrusted);
+  if (bs->certs && certs) {
```

Comment:
Thanks, that makes way more sense. The condition used to fail when I put in `untrusted != NULL`, but when you pointed that out, I realized OpenSSL had written `untrusted` to point to a reference of `bs->certs` or `certs` instead of duplicating it.

Changed the behavior of `untrusted` to point to a copy instead of a pointer reference, and removed the condition since `sk_X509_free` takes `NULL` inputs.